backend/urls.py

```diff
@@ -25,14 +26,17 @@
 from backend.canvas_app_explorer import views as canvas_app_explorer_views #type: ignore
 
 from . import views
+
 # This is for Django Rest Framework
 router = routers.DefaultRouter()
 router.register(r'lti_tools', canvas_app_explorer_views.LTIToolViewSet, basename='ltitool')
 
+# Require being logged in to get to the admin page
+admin.site.login = login_required(admin.site.login)
```

Comment:
So a couple things I noted, some of which may be commented on in the SO thread:
1) This doesn't disable `logout`, but it doesn't break if you use it on the Admin site. I don't see this as a huge issue, but thought I'd raise it as a discussion point.
2) You can still somewhat indirectly surface the login page via redirect. You can observe this if you login as a non-admin user, view the frame source, then navigate to `/admin`. You are logged in, so it doesn't block you from visiting the page, but it knows you don't have the proper permissions and lets you log in as a different user.

Reply:
Yeah, I noticed that if you call that it redirects to login which is disabled. 

If someone did figure out the admin and tried to login, they wouldn't be able to use the page. I don't think this is a risk or anything that's going to be mentioned. It's mostly to just block any potential external login attempts. 

It looks like I probably could just use user_passes_test instead to block it entirely.

`admin.site.login = user_passes_test(False, admin.site.login)`